setup.py

```diff
@@ -55,24 +49,34 @@ def file_walk_relative(top, remove=''):
             yield os.path.join(root, file).replace(remove, '')
 
 
-def std_name_cmd(target_dir):
-    script_path = os.path.join('tools', 'generate_std_names.py')
-    xml_path = os.path.join('etc', 'cf-standard-name-table.xml')
-    module_path = os.path.join(target_dir, 'iris', 'std_names.py')
-    cmd = (sys.executable, script_path, xml_path, module_path)
-    return cmd
+@contextmanager
+def temporary_path(directory):
+    """
+    Context manager that adds and subsequently removes the given directory
+    to sys.path
+
+    """
+    sys.path.insert(0, directory)
+    try:
+        yield
+    finally:
+        del sys.path[0]
+
+
+# Add full path so Python doesn't load any __init__.py in the intervening
+# directories, thereby saving setup.py from additional dependencies.
+with temporary_path('lib/iris/tests/runner'):
+    from _runner import TestRunner  # noqa:
 
 
-class SetupTestRunner(TestRunner, setuptools.Command):
+class SetupTestRunner(TestRunner, Command):
```

Comment:
Was almost tempted to pull the import into this class, and make the runner import entirely lazy...